### PR TITLE
LSP Config for F# using FsAutoComplete

### PR DESCRIPTION
**Pull request by PhilT**
Tested on Ubuntu and Windows.

Ran `make lint`

### CODE REVIEW

File: lua/lspconfig/fsautocomplete.lua [hunk 1]
```diff
@@ -0,0 +1,61 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local bin_path = 'C:/tools/fsac/fsautocomplete.dll'
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:4]: We probably shouldn't be hardcoding these paths

File: lua/lspconfig/fsautocomplete.lua [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+if vim.fn.has('win32') == 0 then
+  bin_path = os.getenv('HOME')..'/lib/fsac/fsautocomplete.dll'
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:6]: Same, this isn't a standard location

File: lua/lspconfig/fsautocomplete.lua [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+end
+
+configs.fsautocomplete = {
+  default_config = {
+    cmd = {'dotnet', bin_path, '--background-service-enabled'};
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:11]: I would just document what this command has to be, and let people populate `cmd`, see sumneko_lua for inspiration

File: lua/lspconfig/fsautocomplete.lua [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    root_dir = util.root_pattern('*.sln', '*.fsproj', '.git');
+    filetypes = {'fsharp'};
+    init_options = {
+      AutomaticWorkspaceInit = true;
+    };
+    settings = {};
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:17]: Do we need to send a blank settings?

File: lua/lspconfig/fsautocomplete.lua [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+Language Server for F# provded by FsAutoComplete (FSAC).
+
+Download a release of FsAutoComplete from [here](https://github.com/fsharp/FsAutoComplete/releases).
+It can also be built from sources. [Instructions](https://github.com/fsharp/FsAutoComplete) are on the main page.
+
+FsAutoComplete requires the [dotnet-sdk](https://dotnet.microsoft.com/download) to be installed.
+
+You may also need to configure the filetype as Vim defaults to Forth for `*.fs` files
+(unless you use something like vim-polyglot):
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:31]: probably obvious but maybe link to vim-polyglot

File: lua/lspconfig/fsautocomplete.lua [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+If you wish to install to a different location set `cmd` as follows:
+
+```lua
+require'lspconfig'.fsautocomplete.setup{
+  cmd = {'dotnet', 'path/to/fsautocomplete.dll', '--background-service-enabled'}
+}
+```
+    ]];
+    default_config = {
+      cmd = [[{'dotnet', '[C:\tools|/$HOME/lib]/fsac/fsautocomplete.dll', '--background-service-enabled'}]];
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:52]: Default config is prepopulated by docgen, so you don't need to duplicate the above settings here

File: lua/lspconfig/fsautocomplete.lua (hunk 4 above)
Comment (PhilT) [reply lua/lspconfig/fsautocomplete.lua]: nope!

File: lua/lspconfig/fsautocomplete.lua (hunk 5 above)
Comment (PhilT) [reply lua/lspconfig/fsautocomplete.lua]: reworded that whole bit to include all the plugins in a separate sentence.

File: lua/lspconfig/fsautocomplete.lua (hunk 6 above)
Comment (PhilT) [reply lua/lspconfig/fsautocomplete.lua]: sweet!

File: lua/lspconfig/fsautocomplete.lua [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+FsAutoComplete requires the [dotnet-sdk](https://dotnet.microsoft.com/download) to be installed.
+
+You may also need to configure the filetype as Vim defaults to Forth for `*.fs` files:
+
+`autocmd BufNewFile,BufRead *.fs,*.fsx,*.fsi set filetype=fsharp`
+
+This is automatically done by plugins such as [vim-polyglot](https://github.com/sheerun/vim-polyglot), [PhilT/vim-fsharp](https://github.com/PhilT/vim-fsharp) or [fsharp/vim-fsharp](https://github.com/fsharp/vim-fsharp).
+
+**By default, this config doesn't have a `cmd` set.** This is because nvim-lspconfig does not make assumptions about your path. You must add the following to your init.vim or init.lua to set `cmd` to the absolute path ($HOME and ~ are not expanded) of you unzipped and compiled fsautocomplete.dll.
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:29]: you -> your

File: lua/lspconfig/fsautocomplete.lua [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    };
+  };
+  docs = {
+    description = [[
+https://github.com/fsharp/FsAutoComplete
+
+Language Server for F# provded by FsAutoComplete (FSAC).
+
+Download a release of FsAutoComplete from [here](https://github.com/fsharp/FsAutoComplete/releases).
+It can also be built from sources. [Instructions](https://github.com/fsharp/FsAutoComplete) are on the main page.
```
Comment (mjlbach) [inline lua/lspconfig/fsautocomplete.lua:19]: Maybe just "instruction to compile from source are found on the main repository.

File: lua/lspconfig/fsautocomplete.lua (hunk 7 above)
Comment (PhilT) [reply lua/lspconfig/fsautocomplete.lua:29]: haha that was copied from the lua one you suggested :)

Comment (mjlbach) [reply lua/lspconfig/fsautocomplete.lua:29]: Bad past me :sad:

**Comment by mjlbach** [comment]
Ok, so tested and g2g?

**Comment by PhilT** [comment]
Yep, looking good.

**Pull request merged**